Dockerfile-base

```diff
@@ -9,56 +9,81 @@ ENV baseDir .
+    \
+# install Node.JS 10
+&&  curl -sSL https://deb.nodesource.com/setup_10.x | bash - \
+&&  apt-get install -y nodejs \
+    \
+&&  rm -rf \
+      /etc/nginx/nginx.conf \
+      /etc/nginx/sites-enabled/default \
+      /var/lib/apt/lists/*
 
 
 # Install qpdf
 # ------------
-WORKDIR /opt
-RUN wget https://s3.amazonaws.com/sharelatex-random-files/qpdf-6.0.0.tar.gz && tar xzf qpdf-6.0.0.tar.gz
-WORKDIR /opt/qpdf-6.0.0
-RUN ./configure && make && make install && ldconfig
-
-
-# Install Node
+RUN mkdir /opt/qpdf \
+    \
+&&  curl -sSL \
+      https://s3.amazonaws.com/sharelatex-random-files/qpdf-6.0.0.tar.gz \
+    | tar -xzC /opt/qpdf --strip-components=1 \
+    \
+&&  cd /opt/qpdf/ \
+&&    ./configure \
+&&    make \
+&&    make install \
+&&    ldconfig \
+    \
+&&  rm -rf /opt/qpdf
+
+
+# Install Grunt
 # ------------
-RUN curl -sL https://deb.nodesource.com/setup_10.x | sudo bash -
-RUN apt-get install -y nodejs
-RUN npm install -g grunt-cli
+RUN npm install -g \
+      grunt-cli \
+&&  rm -rf /root/.npm
 
 
 # Install Node6 (required by some services)
 # -----------------------------------------
-RUN wget https://nodejs.org/dist/v6.17.1/node-v6.17.1-linux-x64.tar.gz && \
-	mkdir -p /opt/nodejs && tar -xzf node-v6.17.1-linux-x64.tar.gz -C /opt/nodejs/ && \
-    cd /opt/nodejs && mv node-v6.17.1-linux-x64 6.17.1 && \
-    ln -s /opt/nodejs/6.17.1/bin/node /usr/bin/node6
+RUN VERSION=6.17.1 \
+&&  mkdir -p /opt/nodejs/${VERSION} \
+&&  curl -sSL \
+      https://nodejs.org/dist/v${VERSION}/node-v${VERSION}-linux-x64.tar.gz \
+    | tar -xzC /opt/nodejs/${VERSION} --strip-components=1 \
+    \
+&&  ln -s /opt/nodejs/${VERSION}/bin/node /usr/bin/node6
 
 
 # Install TexLive
 # ---------------
-RUN wget http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz && \
-	mkdir /install-tl-unx && \
-	tar -xvf install-tl-unx.tar.gz -C /install-tl-unx --strip-components=1
-RUN echo "selected_scheme scheme-basic" >> /install-tl-unx/texlive.profile && \
-	/install-tl-unx/install-tl -profile /install-tl-unx/texlive.profile
-
-# CTAN mirrors occasionally fail, in that case install TexLive against an
-# specific server, for example http://ctan.crest.fr
-# RUN wget http://ctan.crest.fr/tex-archive/systems/texlive/tlnet/install-tl-unx.tar.gz && \
-# 	  mkdir /install-tl-unx && \
-# 	  tar -xvf install-tl-unx.tar.gz -C /install-tl-unx --strip-components=1
-# RUN echo "selected_scheme scheme-basic" >> /install-tl-unx/texlive.profile && \
-# 	  /install-tl-unx/install-tl -profile /install-tl-unx/texlive.profile  \
-# 	  -repository  http://ctan.crest.fr/tex-archive/systems/texlive/tlnet/
-
-RUN rm -r /install-tl-unx; \
-	rm install-tl-unx.tar.gz
-ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/texlive/2019/bin/x86_64-linux/
-RUN tlmgr install latexmk
-RUN tlmgr install texcount
+ARG TEXLIVE_MIRROR=http://mirror.ctan.org/systems/texlive/tlnet
```

Comment:
Can the comment on `CTAN mirrors occasionally fail,...` be added back?

Reply:
Sure. I will add an example shell command with `--build-arg` flags too.